src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp

```diff
@@ -602,9 +622,15 @@ void DistributedAsyncInsertDirectoryQueue::processFilesWithBatching(const Settin
             }
             catch (const Exception & e)
             {
-                if (isDistributedSendBroken(e.code(), e.isRemoteException()))
+                setFilesRetryNum(file_path);
```

Comment:
I also don't see when the retries will be cleared in case of files send properly after some failed attempt.

Reply:
let me add some tests